### ISSUE TITLE
Implement "Script Context"

### ISSUE THREAD

**Issue by bitcraft** [Tuxemon/Tuxemon#598]
For the convenience of map makers and programmers, we should implement a "context" for the script to know various things about when/where the script is called.

Here is a list of things that should be added, please feel free to suggest more:

* the npc/object being interacted with

Python code can then get the specific npc or entity and use that reference in code.  The script could also have actions which can refer to that object without lookups by name.

The "context" object could be implemented as a dictionary which is usable by map action events and conditions.

**Comment by MirceaKitsune** [comment]
This will help resolve the [TODO in NPC wandering](https://github.com/Tuxemon/Tuxemon/blob/development/tuxemon/core/components/event/actions/npc_wander.py#L55) later down the road, by providing a way to check which NPC the player is talking to.

I'm hoping this can be done automatically when interacting with them, instead of requiring an act to set the context. The action lists on map entities can already get pretty big as it is, so the more that can be done internally the better.

**Comment by bitcraft** [comment]
right.  this would be set by the event engine when the script starts.  no actions needed.
